File: article/src/eventPreview.ts
import * as axios from 'axios';
import * as path from 'path';
import * as readline from 'readline';
import { processRequest } from './utils';

const url = 'http://tsugubot.com:8080';
const projectRoot: string = path.resolve(path.dirname(__dirname));


async function main(eventIdString: string) {
    const eventId = parseInt(eventIdString);
    const outputDir = `${projectRoot}/output/eventPreview/${eventId}`

    const promises = [
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewTitle', { eventId }), `${outputDir}/0_eventPreviewTitle_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewDetail', { eventId }), `${outputDir}/1_eventPreviewDetail_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewRules', { eventId }), `${outputDir}/2_eventPreviewRules_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewSongs', { eventId }), `${outputDir}/3_eventPreviewSongs_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewCards', { eventId }), `${outputDir}/4_eventPreviewCards_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewGacha', { eventId }), `${outputDir}/5_eventPreviewGacha_`),
        processRequest(axios.default.post(url + '/eventPreview/eventPreviewCards', { eventId, illustration: true }), `${outputDir}/6_eventPreviewCardIllustration_`)
    ];

    // Wait for all requests to complete
    await Promise.allSettled(promises);
}

// Read eventId from command line
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('请输入活动ID：', (answer) => {
    main(answer);
    rl.close();
});

File: article/pythonSrc/eventPreview.py
import os
import base64
import requests
import asyncio
from aiohttp import ClientSession
import sys

# Constants
url = 'http://tsugubot.com:8080'
project_root = os.path.dirname(__file__)

# Function to process each request
async def process_request(session, endpoint, data, file_path):
    try:
        async with session.post(url + endpoint, json=data) as response:
            response_data = await response.json()
            base64_to_file(response_data, file_path)
    except Exception as e:
        print(e)

# Function to save base64 strings as files
def base64_to_file(base64_list, output_path):
    if not os.path.exists(os.path.dirname(output_path)):
        os.makedirs(os.path.dirname(output_path), exist_ok=True)

    image_id = 1
    for item in base64_list:
        if item['type'] == 'base64':
            image_data = item['string']
            with open(f"{output_path}{image_id}.png", "wb") as file:
                file.write(base64.b64decode(image_data))
                image_id += 1
                print(f"{output_path}{image_id}.png saved successfully!")

# Main function
async def main(event_id_string):
    event_id = int(event_id_string)
    output_dir = f"{project_root}/output/eventPreview/{event_id}"

    async with ClientSession() as session:
        tasks = [
            process_request(session, '/eventPreview/eventPreviewTitle', {'eventId': event_id}, f"{output_dir}/0_eventPreviewTitle_"),
            process_request(session, '/eventPreview/eventPreviewDetail', {'eventId': event_id}, f"{output_dir}/2_eventPreviewDetail_"),
            process_request(session, '/eventPreview/eventPreviewRules', {'eventId': event_id}, f"{output_dir}/3_eventPreviewRules_"),
            process_request(session, '/eventPreview/eventPreviewSongs', {'eventId': event_id}, f"{output_dir}/4_eventPreviewSongs_"),
            process_request(session, '/eventPreview/eventPreviewCards', {'eventId': event_id}, f"{output_dir}/5_eventPreviewCards_"),
            process_request(session, '/eventPreview/eventPreviewGacha', {'eventId': event_id}, f"{output_dir}/6_eventPreviewGacha_"),
            process_request(session, '/eventPreview/eventPreviewCards', {'eventId': event_id, 'illustration': True}, f"{output_dir}/7_eventPreviewCardIllustration_"),
        ]
        await asyncio.gather(*tasks)

# Run the script
if __name__ == "__main__":
    event_id = input("请输入活动ID：")
    asyncio.run(main(event_id))
